azure/scope/machinepool.go

```diff
@@ -535,6 +537,11 @@ func (m *MachinePoolScope) Close(ctx context.Context) error {
 		if err := m.updateReplicasAndProviderIDs(ctx); err != nil {
 			return errors.Wrap(err, "failed to update replicas and providerIDs")
 		}
+		if err := m.updateCustomDataHash(ctx); err != nil {
+			// ignore errors to calculating the custom data hash since it's not absolutely crucial.
+			log.V(4).Error(err, "unable to update custom data hash, ignoring.")
```

Comment:
```suggestion
			log.V(4).Error(err, "unable to update custom data hash, ignoring")
```

Reply:
> // ignore errors to calculating the custom data hash since it's not absolutely crucial.

not sure I agree, we should return all errors as much as possible, so the controller can requeue a new reconcile loop and retry 

this could lead to critical silent failure if the update is failing and causing a user to not be able to scale up a machine pool, or on the other hand throttling if the VMSS is trying to update on every reconcile because its hash annotation is out of date.

If you are concerned about impacting users who don't need this hash (when HasReplicasExternallyManaged is false), then maybe we can move this update inside an if statement to only make the update when it is needed (`if m.HasReplicasExternallyManaged(ctx)`) - this would make sense only if you apply this suggestion https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3134/files#r1114879941

Reply:
Added if block to add a non impact to users not utilizing this feature.